core/src/main/java/org/apache/calcite/rel/rules/ReduceExpressionsRule.java

```diff
@@ -364,14 +366,18 @@ public JoinReduceExpressionsRule(Class<? extends Join> joinClass,
 
     @Override public void onMatch(RelOptRuleCall call) {
       final Join join = call.rel(0);
-      final List<RexNode> expList = Lists.newArrayList(join.getCondition());
+      final RexBuilder rexBuilder = join.getCluster().getRexBuilder();
+      RexSimplify rexSimplify =
```

Comment:
For consistency with the rest of the code in the method, I guess it'd be better if `rexSimplify ` and `condition` are declared `final`

Reply:
Sure~

Reply:
Revised